device/cluster.cpp

```diff
@@ -478,6 +479,30 @@ void Cluster::add_chip(chip_id_t chip_id, std::unique_ptr<Chip> chip) {
     chips_.emplace(chip_id, std::move(chip));
 }
 
+uint32_t Cluster::get_tensix_harvesting_mask(
+    chip_id_t chip_id,
+    tt_ClusterDescriptor* cluster_desc,
+    bool perform_harvesting,
+    std::unordered_map<chip_id_t, uint32_t>& simulated_harvesting_masks) {
+    if (!perform_harvesting) {
+        log_info(LogSiliconDriver, "Skipping harvesting for chip {}.", chip_id);
+        return 0;
+    }
+    uint32_t tensix_harvesting_mask = cluster_desc->get_harvesting_info().at(chip_id);
+    uint32_t simulated_harvesting_mask = (simulated_harvesting_masks.find(chip_id) != simulated_harvesting_masks.end())
+                                             ? simulated_harvesting_masks.at(chip_id)
+                                             : 0;
+    if (simulated_harvesting_mask != 0) {
+        log_info(
+            LogSiliconDriver,
+            "Adding simulated harvesting mask {} for chip {} which has real harvesting mask {}.",
+            simulated_harvesting_mask,
+            chip_id,
+            tensix_harvesting_mask);
+    }
+    return tensix_harvesting_mask | simulated_harvesting_mask;
+}
+
 Cluster::Cluster(
```

Comment:
nit: since we would more often like to do some simulated harvesting, #435, maybe it would be nice to move simulated_harvesting_masks to be before all other parameters. Just maybe, it came across my mind

Reply:
Makes sense but I don't want to introduce API change only because of this.